test/bin/generate_common_versions.py

```diff
@@ -0,0 +1,253 @@
+"""
+The generate_common_versions.py generates all variables for the common_versions.sh script and prints them to stdout.
+"""
+import requests
+import subprocess
+import os
+
+ARCH = os.uname().machine
+
+MINOR = 19
+
+
+def get_candidate_repo(minor, dev_preview=False):
+    """
+    :param minor: the minor version, e.g. 19 for 4.19
+    :param dev_preview: if True, returns the engineering candidate repo, otherwise returns the release candidate repo
+
+    ``get_candidate_repo`` returns the URL of the EC/RC repository for the specified minor version.
+    """
+    if dev_preview:
+        return f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/microshift/ocp-dev-preview/latest-4.{minor}/el9/os"
+    else:
+        return f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/microshift/ocp/latest-4.{minor}/el9/os"
+
+
+def get_beta_repo(minor):
+    """
+    :param minor: the minor version, e.g. 19. for 4.19
+
+    ``get_beta_repo`` returns the URL of the beta repository for the specified minor version.
+    If the repo for the wanted version does not exist or it does not provide the necessary
+    packages, it looks for previous releases, for up to three previous minor versions.
+    """
+    for i in range(minor, minor-3, -1):
+        url = f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/dependencies/rpms/4.{i}-el9-beta"
+        if repo_exists(url) and provides_pkg(url, "cri-o"):
+            return f'"{url}"'
```

Comment:
I think we could add quotes as last step, when rendering the file and otherwise keep it clean
```suggestion
            return url
```

Reply:
Done